docs/source/core_specification/schema_types.rst

```diff
@@ -1,3 +1,282 @@
+    }
+
+    union Union__Food__Species = Food | Species
+
+
+.. note::
+
+    A GraphQL schema can be serialized with the :code:`print_schema` function from the
+    :code:`graphql.utils.schema_printer` module.
+
+
+Objects and Fields
+--------------------------
+
+The core components of a GraphQL schema are GraphQL object types. They conceptually represent the
+concrete vertices in the underlying database. For relational databases, we think of the tables as
+the concrete vertices.
+
+Lets go over a toy example of a GraphQL object type:
+
+.. code::
+
+    type Toy {
+        name: String
+        in_Animal_PlaysWith: [Animal]
+    }
+
+Here are some of the details:
+
+- :code:`name` is a **property field** that represents concretes data.
+- :code:`in_Animal_PlaysWith` is a **vertex field** representing an outbound edge.
+- :code:`String` is a built-in GraphQL scalar type.
+- :code:`[Animal]` is a GraphQL list type that represents an array of :code:`Animal`
+  objects.
+
+
+Directives
+----------
+
+In this section we'll go over how query directives are defined. For information on the available
+query directives and their semantics see :doc:`query directives <query_directives>`.
+
+Let's look at the :code:`@output` directive:
+
+.. code::
+
+   directive @output(out_name: String!) on FIELD
+
+- :code:`@output` defines the directive name.
+- :code:`out_name: String!` is a GraphQL argument. The :code:`!` indicates that it must not be null.
+- :code:`on FIELD` defines where the directive can be located. According to the definition, this
+  directive can only be located next to fields. The compiler might have additional restrictions
+  for where a query can be located. See :doc:`query directives <query_directives>` for more info.
+
+Scalars
+-------
+
+The compiler uses the built-in GraphQL
+`scalar types <https://graphql.org/learn/schema/#scalar-types>`__ as well as three custom scalars:
+
+- :code:`DateTime` represents timezone-aware second-accuracy timestamps.
+- :code:`Date` represents day-accuracy date objects.
+- :code:`Decimal` is an arbitrary-precision decimal number object useful for representing values
+  that should never be rounded, such as currency amounts.
+
+Query Operation
+---------------
+
+GraphQL allows for three operation types *query*, *mutation* and *subscription*. The compiler
+only allows query operation types as shown in the code snippet below:
+
+.. code::
+
+    schema {
+        query: RootSchemaQuery
+    }
+
+A query may begin in any of the **root vertex types** specified by the :code:`RootSchemaQuery`
+object:
+
+.. code::
+
+    type RootSchemaQuery {
+        Animal: [Animal]
+        Entity: [Entity]
+        Food: [Food]
+        Species: [Species]
+        Toy: [Toy]
+    }s
+
+Inheritance
+-----------
+
+The compiler uses GraphQL interfaces and GraphQL unions in representing the inheritance structure
+of the underlying schema. You may ignore this section if are compiling to a language with no
```

Comment:
Instead of `You may ignore this section`, say that not all databases support inheritance, so this is only a supported feature on some backends.